gui/src/hooks/firmware-tool.ts

```diff
@@ -0,0 +1,129 @@
+import { createContext, useContext, useState } from 'react';
+import { fetchGetFirmwaresDefaultConfigBoard, useGetHealth } from '../firmware-tool-api/firmwareToolComponents';
+import { BuildResponseDTO, CreateBoardConfigDTO, CreateBuildFirmwareDTO, DefaultBuildConfigDTO } from '../firmware-tool-api/firmwareToolSchemas';
+import { BoardPinsForm } from '../components/firmware-tool/BoardPinsStep';
+import { DeepPartial } from 'react-hook-form';
+import { BoardType, FirmwareUpdateStatus, FlashingMethod } from 'solarxr-protocol';
+
+export type PartialBuildFirmware = DeepPartial<CreateBuildFirmwareDTO>
+export type FirmwareBuildStatus = BuildResponseDTO;
+export type SelectedDevice = { type: FlashingMethod, deviceId: string | number, deviceNames: string[] };
+
+
+export const boardTypeToFirmwareToolBoardType: Record<BoardType, CreateBoardConfigDTO['type'] | null> = {
```

Comment:
You could import the firmware protocol package from NPM 😉 
https://www.npmjs.com/package/@slimevr/firmware-protocol